### ISSUE TITLE
too many arguments to ki_complete

### ISSUE THREAD

**Issue by yuanjin0928** [hmaarrfk/dma_ip_drivers#10]
Hi,

I want to compile the dma_ip_drivers on my laptop whose linux kernel version is 5.14.0-284.25.1.el9_2.x86_64.
However, I got the following error:
my_path/dma_ip_drivers/XDMA/linux-kernel/xdma/cdev_sgdma.c:108:17: error: too many arguments to function ‘caio->iocb->ki_complete’
      caio->iocb->ki_complete(caio->iocb, res, res2);
What I am confused is that the number of parameters of ki_complete is reduced to 2 since kernel 5.16, but it does have 3 arguments for kernel 5.14. Do you know what is the potential reason?

Regards,
Jin

**Comment by hmaarrfk** [comment]
I remember that the parameter was unused. 

for QDMA I guard it with compiler directives and switch accordingly.

**Comment by yuanjin0928** [comment]
Yes, I have already seen some updates in the official xilinx github repository. And I am planning to modify the code.
But what makes me confused is that my linux kernel's version is 5.14.0, the ki_complete function should have three arguments passed in. However, when I check the fs.h (ki_complete is declared in this file) in my /usr/src/kernels/5.14.0-284.25.1.el9_2.x86_64/include/linux/ directory, I find ki_complete only has two parameters. I also check the fs.h in the /lib/modules/5.14.0-284.25.1.el9_2.x86_64/buil/include/linux/ directory, it also has two parameter. I have no idea what is going on.

In addition, in the source file, a lot of files include directives like "#include <linux/*.h>". Does anyone know where gcc looks for these header files. I run the command "cpp -v", it shows:
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-redhat-linux/11/include
 /usr/local/include
 /usr/include
End of search list.
Obviously, it doesn't include the directories in which I find the fs.h file.

**Comment by hmaarrfk** [comment]
i know redhat backports things
https://github.com/hmaarrfk/dma_ip_drivers/blob/64e7fda6d1c42f8b93d8e0391bb17cc880ed5308/XDMA/linux-kernel/xdma/libxdma.h#L37

**Comment by yuanjin0928** [comment]
> i know redhat backports things
> 
> https://github.com/hmaarrfk/dma_ip_drivers/blob/64e7fda6d1c42f8b93d8e0391bb17cc880ed5308/XDMA/linux-kernel/xdma/libxdma.h#L37

It looks like what I need to know. Thank you very much. I will check it immediately.
